examples/composite_dialog/my_app_window/mod.rs

```diff
@@ -0,0 +1,80 @@
+mod imp;
+
+use gtk::{
+    glib::{self, ParamSpec},
+    prelude::*,
+    ResponseType,
+};
+
+glib::wrapper! {
+    pub struct MyAppWindow(ObjectSubclass<imp::MyAppWindow>)
+        @extends gtk::Widget, gtk::Window, gtk::ApplicationWindow;
+}
+
+#[gtk::template_callbacks]
+impl MyAppWindow {
+    pub fn new<P: glib::IsA<gtk::Application>>(app: &P) -> Self {
+        glib::Object::builder().property("application", app).build()
+    }
+
+    /// Callback handler for notify::label signal.
+    ///
+    /// When counter property reach 3, a dialog pops up and present the user
+    /// with 2 choices: Set the counter to 6 or reset the counter to 0.
+    #[allow(deprecated)]
```

Comment:
I would leave a comment explaining what is deprecated: gtk::Dialog but see my further comments first